libs/community/langchain_community/chat_message_histories/dynamodb.py

```diff
@@ -167,16 +167,19 @@ def add_message(self, message: BaseMessage) -> None:
                 import time
 
                 expireAt = int(time.time()) + self.ttl
-                self.table.put_item(
-                    Item={
-                        **self.key,
-                        self.history_messages_key: messages,
-                        self.ttl_key_name: expireAt,
-                    }
+                self.table.update_item(
+                    Key={**self.key},
+                    UpdateExpression=(
+                        f"set {self.history_messages_key} = :h, "
```

Comment:
Does this value need to be sanitized to avoid query injection?  We can validate it at the __init__ or here

same for ttl_key_name and anything else that might require escaping?

Reply:
Thanks for the review!

For query injection, I am confident the Boto API here will prevent it. The API builds the query, we don't pass one in, so you can see the `UpdateExpression` just sets the update action and parameterizes the fields to apply the action to, and the `ExpressionAttributeValues` maps the values to the parameters. While the update expression could be tampered with, the most you could do is pass a `self.history_messages_key` that is composed of additional actions and attribute names. But the value of `self.history_messages_key` is set by the developer and is not exposed to end-user data. Only the chat history items passed in `ExpressionAttributeValues` are exposed to user data and that's already parameterized, so having a DynamoDB query in your chat history for example will just send that query as a value in your chat history, it has no way of being executed. Same goes for TTL.